lib/std/Build/Step/Compile.zig

```diff
@@ -64,6 +64,8 @@ initial_memory: ?u64 = null,
 max_memory: ?u64 = null,
 shared_memory: bool = false,
 global_base: ?u64 = null,
+/// Tells the linker to not output an entry point.
+no_entry: ?bool = null,
```

Comment:
I think this can be merged with `entry_symbol_name`, something like this:

```zig
entry: Entry,

const Entry = union(enum) {
    /// Let the compiler decide whether to make an entry point and what to name
    /// it.
    default,
    /// The executable will have no entry point.
    disabled,
    /// The executable will have an entry point with the default symbol name.
    enabled,
    /// The executable will have an entry point with the specified symbol name.
    symbol_name: []const u8,
};
```